x-pack/plugins/lens/public/xy_visualization_plugin/xy_config_panel.test.tsx

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { XYConfigPanel } from './xy_config_panel';
+import { DatasourcePublicAPI } from '../types';
+import { XYArgs, SeriesType } from './types';
+import { Position } from '@elastic/charts';
+
+describe('XYConfigPanel', () => {
+  function mockDatasource(): DatasourcePublicAPI {
+    return {
+      duplicateColumn: () => [],
+      getOperationForColumnId: () => null,
+      generateColumnId: () => 'TESTID',
+      getTableSpec: () => [],
+      moveColumnTo: () => {},
+      removeColumnInTableSpec: () => [],
+      renderDimensionPanel: () => {},
+    };
+  }
+
+  function testState(): XYArgs {
+    return {
+      legend: { isVisible: true, position: Position.Right },
+      seriesType: 'bar',
+      splitSeriesAccessors: [],
+      stackAccessors: [],
+      title: 'Test Chart',
+      x: {
+        accessor: 'foo',
+        position: Position.Bottom,
+        showGridlines: true,
+        title: 'X',
+      },
+      y: {
+        accessors: ['bar'],
+        position: Position.Left,
+        showGridlines: true,
+        title: 'Y',
+      },
+    };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function testSubj(component: ReactWrapper<any>, subj: string) {
+    return component
+      .find(`[data-test-subj="${subj}"]`)
+      .first()
+      .props();
+  }
+
+  test('toggles axis position when going from horizontal bar to any other type', () => {
+    const changeSeriesType = (fromSeriesType: SeriesType, toSeriesType: SeriesType) => {
+      const setState = jest.fn();
+      const state = testState();
+      const component = mount(
+        <XYConfigPanel
+          datasource={mockDatasource()}
+          setState={setState}
+          state={{ ...state, seriesType: fromSeriesType }}
+        />
+      );
+
+      (testSubj(component, 'lnsXY_seriesType').onChange as Function)(toSeriesType);
+
+      expect(setState).toHaveBeenCalledTimes(1);
```

Comment:
Could you make a stronger assertion than "called once"? Could you assert that it was called with a new axis position? This applies to most of your tests here, I don't think they're strong enough.

Reply:
I am checking the actual call further down (in the expect statements). See lines 75 - 97.